### PR TITLE
Added isExpired logic for UrlEntity

### PR DESCRIPTION
**Pull request by dapariciom**
Found some potential fixes and added isExpired logic

### CODE REVIEW

File: src/main/java/com/example/shorturl/controller/api/ShortenUrlController.java [hunk 1]
```diff
@@ -39,15 +38,13 @@ public ShortenUrlController(final ShortenUrlService shortenUrlService){
...
 
-        if(Objects.isNull(urlRequest) || StringUtils.isEmpty(urlRequest.getUrl())){
+        if(Objects.isNull(urlRequest) || StringUtils.isEmpty(urlRequest.getUrl()))
             throw new ResponseStatusException(HttpStatus.BAD_REQUEST, "Url request is missing or empty");
-        }
 
         Optional<UrlEntity> optionalUrl = shortenUrlService.shortUrl(urlRequest);
 
-        if(optionalUrl.isEmpty()){
+        if(optionalUrl.isEmpty())
```
Comment (BeeAM17) [inline src/main/java/com/example/shorturl/controller/api/ShortenUrlController.java]: UrlEntity entity = optionalUrl.orElseThrow(() -> new ResponseStatusException(HttpStatus.BAD_REQUEST, "Url is missing"));

File: src/main/java/com/example/shorturl/controller/api/ShortenUrlController.java [hunk 2]
```diff
@@ -86,13 +83,18 @@ public ResponseEntity<UrlResponse> redirectUrl(@PathVariable String shortLink, H
     }
 
     @GetMapping("/{id}")
-    public ResponseEntity<Optional<UrlEntity>> findById(@PathVariable Integer id){
+    public ResponseEntity<UrlEntity> findById(@PathVariable Integer id) throws UrlNotFoundException {
+
+        Optional<UrlEntity> optionalUrl = shortenUrlService.findById(id);
+
+        if(optionalUrl.isEmpty())
```
Comment (BeeAM17) [inline src/main/java/com/example/shorturl/controller/api/ShortenUrlController.java]: Same here

**Review by BeeAM17 — CHANGES_REQUESTED** [review]

File: src/main/java/com/example/shorturl/controller/api/ShortenUrlController.java (hunk 1 above)
Comment (dapariciom) [reply src/main/java/com/example/shorturl/controller/api/ShortenUrlController.java]: Fixed

File: src/main/java/com/example/shorturl/controller/api/ShortenUrlController.java (hunk 2 above)
Comment (dapariciom) [reply src/main/java/com/example/shorturl/controller/api/ShortenUrlController.java]: Fixed

**Review by BeeAM17 — APPROVED** [review]

**Pull request merged**